### ISSUE TITLE
Make JsonField configurable in parallel with codemirror api

### ISSUE THREAD

**Issue by febbraiod** [exhi/hoist-react#145]
For example, in some contexts we want line-wrapping and in some we do not. This particular change was implemented as part of the addition of the client error and activity detail popups. After discussing with Lee we decided that if we are going to open up this component to customization we should do so more broadly.

Probably we should implement a codeMirrorConfig object that can be passed in and merged with a set of defaults. In the interest of not holding up the popup branch, I am creating this as a separate ticket.
